pkg/node/node.go

```diff
@@ -218,36 +219,52 @@ func NewBee(addr string, publicKey *ecdsa.PublicKey, signer crypto.Signer, netwo
 	addressbook := addressbook.New(stateStore)
 
 	var (
-		swapBackend        transaction.Backend
+		swapChain          transaction.Backend = &swapChainTransactionStub{logger}
 		overlayEthAddress  common.Address
 		chainID            int64
 		transactionService transaction.Service
 		transactionMonitor transaction.Monitor
 		chequebookFactory  chequebook.Factory
-		chequebookService  chequebook.Service
+		chequebookService  chequebook.Service = &chequebookServiceStub{}
 		chequeStore        chequebook.ChequeStore
 		cashoutService     chequebook.CashoutService
 		pollingInterval    = time.Duration(o.BlockTime) * time.Second
 	)
-	swapBackend, overlayEthAddress, chainID, transactionMonitor, transactionService, err = InitChain(
-		p2pCtx,
-		logger,
-		stateStore,
-		o.SwapEndpoint,
-		signer,
-		pollingInterval,
-	)
-	if err != nil {
-		return nil, fmt.Errorf("init chain: %w", err)
+
+	var swapBackendEnabled = !o.StubSwapChain || // Will stay enabled only in LightNode mode.
```

Comment:
this looks like possible error prone check... how does this correspond with the `swap-enable` flag? maybe better to have some switch case here or some more verbose conditional?

Reply:
changed to a switch